src/App.tsx

```diff
@@ -1,14 +1,55 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[] | []>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [filterType, setFilterType] = useState('all');
+
+  const selectTodo = (todoId: number) => {
+    setSelectedTodo(todos.find(todo => todo.id === todoId) || null);
+  };
+
+  const closeModal = () => setSelectedTodo(null);
+
+  const filterTodos = () => {
+    const cleanQuery = query.trim().toLowerCase();
+
+    switch (filterType) {
+      case 'active':
+        return todos.filter(todo => !todo.completed && todo.title.toLowerCase().includes(cleanQuery));
+      case 'completed':
+        return todos.filter(todo => todo.completed && todo.title.toLowerCase().includes(cleanQuery));
+      default:
+        return todos.filter(todo => todo.title.toLowerCase().includes(cleanQuery));
```

Comment:
maybe let's create separate function for `todos.filter(todo => !todo.completed && todo.title.toLowerCase().includes(cleanQuery));`?